Avoid duplicate parametrizations in a prominent `SkyCoord` roundtripping test

eerovaher opened the pull request:
### Description

There are roughly 2000 unit tests in `coordinates`, 256 of which are parametrizations of the single test I am updating here. But 128 parametrizations are useless and can be removed without loss of test coverage. On current `main`:
```
$ pytest astropy/coordinates/
  ...
============================ 2004 passed, 50 skipped, 12 xfailed in 13.67s ============================

```
With this PR:
```
$ pytest astropy/coordinates/
  ...
============================ 1876 passed, 50 skipped, 12 xfailed in 12.89s ============================

```

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

mhvk reviewed (commented):
I like very much the idea of speeding up those tests, but ideally without duplicating the code. Can we not simply avoid generating the code sets that end up repeating the tests? I.e., in the nested `for` loops, iterate over multiple `equinox1` only if the relevant frame has an `equinox` argument?

eerovaher committed:
Avoid bad parametrizations of a prominent test

Parametrizations of a single `SkyCoord` roundtripping test made up
roughly 1/8 of all `coordinates` unit tests, but half of those
parameterizations were unnecessary. Rewriting the parametrization setup
avoids the unnecessary parametrizations and replacing parametrization of
one of the variables with 4 different tests simplifies the
implementation.
